Nucleo_h743zi fails to format storage flash partition

**Describe the bug**
The file system fails to format the storage partition on the nucleo_h743zi board. This is failing on both the `samples/subsys/shell/fs` and the `samples/subsys/fs/littlefs`. Also one thing to note is the docs for the littlefs example say that it was created using the nucleo_h743zi board which us the same board I am attempting to test with.

**To Reproduce**

***littlefs Example***
1. `west build -b nucleo_h743zi zephyr/samples/subsys/fs/littlefs -p auto`
2. Build fails with the following error code:
```
[4/16] Building C object CMakeFiles/app.dir/src/main.c.obj
FAILED: CMakeFiles/app.dir/src/main.c.obj 
ccache /home/jowen/Downloads/zephyr-sdk-0.15.1/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DCORE_CM7 -DHSE_VALUE=8000000 -DKERNEL -DLFS_CONFIG=zephyr_lfs_config.h -DSTM32H743xx -DUSE_FULL_LL_DRIVER -DUSE_HAL_DRIVER -D_FORTIFY_SOURCE=2 -D__PROGRAM_START -D__ZEPHYR__=1 -I/home/jowen/dev/zephyrproject/zephyr/include -I/home/jowen/dev/zephyrproject/build/zephyr/include/generated -I/home/jowen/dev/zephyrproject/zephyr/soc/arm/st_stm32/stm32h7 -I/home/jowen/dev/zephyrproject/zephyr/drivers -I/home/jowen/dev/zephyrproject/zephyr/soc/arm/st_stm32/common -I/home/jowen/dev/zephyrproject/modules/hal/cmsis/CMSIS/Core/Include -I/home/jowen/dev/zephyrproject/modules/hal/atmel/include -I/home/jowen/dev/zephyrproject/zephyr/modules/hal_nxp/. -I/home/jowen/dev/zephyrproject/modules/hal/stm32/stm32cube/stm32h7xx/soc -I/home/jowen/dev/zephyrproject/modules/hal/stm32/stm32cube/stm32h7xx/drivers/include -I/home/jowen/dev/zephyrproject/modules/hal/stm32/stm32cube/stm32h7xx/drivers/include/Legacy -I/home/jowen/dev/zephyrproject/modules/hal/stm32/stm32cube/common_ll/include -I/home/jowen/dev/zephyrproject/modules/fs/littlefs -I/home/jowen/dev/zephyrproject/zephyr/modules/littlefs -isystem /home/jowen/dev/zephyrproject/zephyr/lib/libc/minimal/include -isystem /home/jowen/Downloads/zephyr-sdk-0.15.1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/include -isystem /home/jowen/Downloads/zephyr-sdk-0.15.1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/include-fixed -fno-strict-aliasing -Og -imacros /home/jowen/dev/zephyrproject/build/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m7 -mthumb -mabi=aapcs -mfp16-format=ieee --sysroot=/home/jowen/Downloads/zephyr-sdk-0.15.1/arm-zephyr-eabi/arm-zephyr-eabi -imacros /home/jowen/dev/zephyrproject/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/home/jowen/dev/zephyrproject/zephyr/samples/subsys/fs/littlefs=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/jowen/dev/zephyrproject/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/jowen/dev/zephyrproject=WEST_TOPDIR -ffunction-sections -fdata-sections -std=c99 -nostdinc -MD -MT CMakeFiles/app.dir/src/main.c.obj -MF CMakeFiles/app.dir/src/main.c.obj.d -o CMakeFiles/app.dir/src/main.c.obj -c /home/jowen/dev/zephyrproject/zephyr/samples/subsys/fs/littlefs/src/main.c
In file included from /home/jowen/dev/zephyrproject/zephyr/include/zephyr/arch/arm/aarch32/arch.h:20,
                 from /home/jowen/dev/zephyrproject/zephyr/include/zephyr/arch/cpu.h:19,
                 from /home/jowen/dev/zephyrproject/zephyr/include/zephyr/kernel_includes.h:33,
                 from /home/jowen/dev/zephyrproject/zephyr/include/zephyr/kernel.h:17,
                 from /home/jowen/dev/zephyrproject/zephyr/samples/subsys/fs/littlefs/src/main.c:12:
/home/jowen/dev/zephyrproject/zephyr/include/zephyr/devicetree.h:190:36: error: 'DT_N_NODELABEL_storage_partition_PARTITION_ID' undeclared here (not in a function)
  190 | #define DT_NODELABEL(label) DT_CAT(DT_N_NODELABEL_, label)
      |                                    ^~~~~~~~~~~~~~~
/home/jowen/dev/zephyrproject/zephyr/include/zephyr/devicetree.h:3714:24: note: in definition of macro 'DT_CAT'
 3714 | #define DT_CAT(a1, a2) a1 ## a2
      |                        ^~
/home/jowen/dev/zephyrproject/zephyr/include/zephyr/storage/flash_map.h:292:35: note: in expansion of macro 'DT_FIXED_PARTITION_ID'
  292 | #define FIXED_PARTITION_ID(label) DT_FIXED_PARTITION_ID(DT_NODELABEL(label))
      |                                   ^~~~~~~~~~~~~~~~~~~~~
/home/jowen/dev/zephyrproject/zephyr/include/zephyr/devicetree.h:190:29: note: in expansion of macro 'DT_CAT'
  190 | #define DT_NODELABEL(label) DT_CAT(DT_N_NODELABEL_, label)
      |                             ^~~~~~
/home/jowen/dev/zephyrproject/zephyr/include/zephyr/storage/flash_map.h:292:57: note: in expansion of macro 'DT_NODELABEL'
  292 | #define FIXED_PARTITION_ID(label) DT_FIXED_PARTITION_ID(DT_NODELABEL(label))
      |                                                         ^~~~~~~~~~~~
/home/jowen/dev/zephyrproject/zephyr/samples/subsys/fs/littlefs/src/main.c:271:32: note: in expansion of macro 'FIXED_PARTITION_ID'
  271 |         .storage_dev = (void *)FIXED_PARTITION_ID(storage_partition),
      |                                ^~~~~~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /usr/bin/cmake --build /home/jowen/dev/zephyrproject/build
```

***shell/fs Example***
1. `west build -b nucleo_h743zi zephyr/samples/subsys/shell/fs -p auto`
2. `west flash`
3. Send the following command over the uart console to the nucleo. `fs mount littlefs /lfs`
4. Mount command fails:
```
*** Booting Zephyr OS build zephyr-v3.2.0  ***

[00:00:00.000,000] <dbg> fs: fs_register: fs register 1: 0
uart:~$ fs mount littlefs /lfs
Error mounting as littlefs: -28
[00:00:19.609,000] <inf> littlefs: LittleFS version 2.4, disk version 2.0
[00:00:19.609,000] <dbg> littlefs: littlefs_flash_init: FS area 1 at 0x20000 for 131072 bytes
[00:00:19.610,000] <inf> littlefs: FS at flash-controller@52002000:0x20000 is 1 0x20000-byte blocks with 512 cycle
[00:00:19.610,000] <inf> littlefs: sizes: rd 16 ; pr 16 ; ca 64 ; la 32
[00:00:19.610,000] <wrn> littlefs: can't mount (LFS -84); formatting
[00:00:19.610,000] <err> littlefs: WEST_TOPDIR/modules/fs/littlefs/lfs.c:567: No more free space 1
[00:00:19.610,000] <err> littlefs: format failed (LFS -28)
[00:00:19.610,000] <err> fs: fs mount error (-28)
uart:~$ 
```

**Environment (please complete the following information):**
OS: Ubuntu 20.04
Toolchain Zephyr SDK 0.15.1
Commit SHA: https://github.com/zephyrproject-rtos/zephyr/commit/4256cd41df6c60f1832fd2deb14edc30ac7debab


https://github.com/zephyrproject-rtos/zephyr/issues/51870#tasklist-block-f097606c-01c2-406a-a4fa-b86cd95f4d8c

--- a/samples/subsys/fs/littlefs/boards/nucleo_h743zi.overlay
+++ b/samples/subsys/fs/littlefs/boards/nucleo_h743zi.overlay
@@ -38,9 +38,9 @@
 			   #address-cells = <1>;
 			   #size-cells = <1>;
 
-			   partition@0 {
+			   storage_partition: partition@0 {
 			       label = "storage";
-			       reg = <0x01000000 DT_SIZE_M(8)>;
+			       reg = <0 DT_SIZE_M(8)>;
 			   };
 		};
 	};

